src/cheri_sys_exceptions.sail

```diff
@@ -47,6 +51,18 @@ function prepare_trap_vector(p : Privilege, c : Mcause) -> xlenbits = {
                            Supervisor => STCC,
                            User       => UTCC
                          };
+
+  /*
+   * XXX I'd prefer to use match here, but I don't understand how to shift
+   * types in the other direction, that is, to go from typeof(tcc.otype)
+   * to typeof(otype_unsealed).
+   */
+  let tcc = if      tcc.otype == to_bits(otype_width, otype_unsealed)
```

Comment:
I wonder if we should add helpers such as `capIsSentry`/`capIsUnsealed`/`capIsSealedWithType` (or similar) to cheri_cap_common.sail? Not sure about the naming conventions there, seems quite inconsistent.

Reply:
xTCC is fundamentally incompatible with sentries given you have a vectored mode. In theory you could support sentries for direct mode, as the low bits happen to be 0 so it's taken as-is, but this is the only possible exception and I don't particularly like that conditional behaviour.

Reply:
Ah, fair.  I'll drop this hunk.